### ISSUE TITLE
Cannot compile upm properly anymore, cannot get new modules (Solved-workaround)

### ISSUE THREAD

**Issue by g-vidal** [intel-iot-devkit/upm#379]
Hi I have got a strange problem

I did a git pull from intel-iot-devkit/upm  on my Edison and when I try to compile I get errors  like this :
` Nothing known about base class 'ITemperatureSensor'. Ignored.`
The python modules that succeed arrive there :
`/usr/lib/python./site-packages/`
instead of  :
`/usr/lib/python2.7/site-packages/`
I did not change anything in the process
```shell
cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr ..
make
sudo make install
```
I cannot figure out what happens for the BMP180 that I need and more generally for the modules that do not appear :
```python
>>> import pyupm_bmp180
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named pyupm_bmp180
```

Some more information 
When installing I get systematically the message 
```shell
-- Installing: /usr/lib/python./site-packages/_pyupm_grovewfs.so
-- Installing: /usr/lib/python./site-packages/pyupm_grovewfs.py
```
Why is it trying to install in `/usr/lib/python.` instead of  `/usr/lib/python2.7` which exists and works perfectly.
Any help would be greatly appreciated, I have prepared a training session and nothing works any more ;-(
Thanks

**Comment by xbolshe** [comment]
It seems that ${PYTHON_VERSION_MAJOR} and ${PYTHON_VERSION_MINOR} are not detected.

**Comment by g-vidal** [comment]
Follow-up....

I deadly need to be able to compile mraa and ump libs with students on various platforms : Edison / RaspBerry-Pi  PClinux Debian, then I have set up a workaround:

- As suggested by @xbolshe   I wrote fixed values for  ${PYTHON_VERSION_MAJOR} and ${PYTHON_VERSION_MINOR}  in upm/src/CMakeLists.txt 
- I removed all declarations like public ILightSensor, public upm::ILightController, public ITemperatureSensor, public IHumiditySensor, public IPressureSensor that introduce warnings and sometimes compilation errors
- I think that for multiple reasons including science teacher responsability we need to spell properly scientists names then I wrote properly Anders Celsius name included in some functions.

I have not had time to check if the functions were properly working but they compile and at least they can be imported from python prompt ! If anybody is interested the workaround is there : [on branch upm+GV](https://github.com/g-vidal/upm.git).
If anybody with better skills than me can bring back upm lib in the original fluent compilation mood it will be greatly appreciated ;-).

**Comment by Propanu** [comment]
Python issues should be fixed now with e8090161522962fff3ef38ea8445f9cfac49cd24. Many new features were introduced with version 0.6.0, and unfortunately we still need to ensure compatibility with all the possible build combinations. When using this in a sensitive environment (academia or production) I highly recommend sticking to a stable version (v.0.5.1) until the new release is reinforced in case you encounter build errors. Thanks a lot for your feedback and patience!

**Comment by g-vidal** [comment]
Many many thanks for your work !

Le 29/03/2016 19:02, Mihai Tudor Panu a écrit :
>
> Python issues should be fixed now with e809016
> <https://github.com/intel-iot-devkit/upm/commit/e8090161522962fff3ef38ea8445f9cfac49cd24>.
> Many new features were introduced with version 0.6.0, and
> unfortunately we still need to ensure compatibility with all the
> possible build combinations. When using this in a sensitive
> environment (academia or production) I highly recommend sticking to a
> stable version (v.0.5.1) until the new release is reinforced in case
> you encounter build errors.
>
yes I know :-[ :-[  You are right  I should  butO:-) ..... It is  so fun
to ride the wave....8-)
>
> Thanks a lot for your feedback and patience!
>
With pleasure, in my opinion the important point is to teach how to make
it work, in this cas eunfortunately the problem was at the very
beginning of the process... I did not notice that we jumped from 0.5 to
0.6....
I will check again and let you know.

Best regards
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/intel-iot-devkit/upm/issues/379#issuecomment-202999226>
>

-- 
Gérard Vidal
Chargé de mission normes et veille technologique et numériques auprès du
Directeur de l'IFÉ.
Université de Lyon : IFÉ / ENS de Lyon 15 Parvis rené Descartes 69342
Lyon Cedex07.
tel : [+33] (0)4 26 73 12 60.
onglet logos

**Comment by Propanu** [comment]
Closing, as v0.6.1 should fix all these issues. Thanks again!
